controllers/reconcilers/configuration/configuration_reconciler.go

```diff
@@ -793,3 +793,49 @@ func (r *Reconciler) waitForResourcesDeployment(ctx context.Context, cr *v1.Obse
 	}
 	return v1.ResultInProgress, nil
 }
+
+func (r *Reconciler) GetOauth2ConfigSecret(ctx context.Context, cr *v1.Observability) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	selector := client.ObjectKey{
+		Name:      RemoteWriteOIDCSecretName,
+		Namespace: cr.Namespace,
+	}
+
+	err := r.client.Get(ctx, selector, secret)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+func (r *Reconciler) GetRemoteWriteOauth2Config(ctx context.Context, cr *v1.Observability) (*prometheusv1.OAuth2, error) {
+	secret, err := r.GetOauth2ConfigSecret(ctx, cr)
```

Comment:
Should we return nil config if the secret is nil?

Reply:
Yes, good point